src/grammar/grammar.ts

```diff
@@ -1012,6 +1013,31 @@ semantics.addOperation<AstNode>("astOfStatement", {
             loc: createRef(this),
         });
     },
+    StatementDestruct(
+        _letKwd,
+        typeId,
+        _lparen,
+        identifiers,
+        _optTrailingComma,
+        _rparen,
+        _equals,
+        expression,
+        _semicolon,
+    ) {
+        return createAstNode({
+            kind: "statement_destruct",
+            type: typeId.astOfType(),
+            identifiers: identifiers
+                .asIteration()
+                .children.reduce((map, item) => {
+                    const destructItem = item.astOfExpression();
+                    map.set(destructItem.field, destructItem.name);
```

Comment:
looks like the following 
```tact
struct S { x: Int; y: String; z: Bool }
let S { x: a, x: b } = s;
```

will get past the typechecker (but it should trigger an error)